src/lib/CLValue/utils.ts

```diff
@@ -156,68 +158,71 @@ export const matchTypeToCLType = (type: any): CLType => {
 export const matchByteParserByCLType = (
   val: CLType
 ): Result<CLValueBytesParsers, string> => {
-  if (val.tag === CLTypeTag.Bool) {
+  if (val.linksTo === ACCOUNT_HASH_ID) {
```

Comment:
This file looks like a big switch whether than successive if statements. I don't believe this style would pass a code smell sniffer like Sonarcloud. I would suggest to ask for a subscription at Sonarcloud to Ed and pass the whole plugin into their analyser. That much retunrs in a function would probably not be accepted.

Reply:
I changed it to `switch`, there is still one place where I needed to use `ifs` but hopefully this can be refactored someday...